src/hotspot/share/opto/loopopts.cpp

```diff
@@ -367,6 +352,19 @@ void PhaseIdealLoop::dominated_by(IfProjNode* prevdom, IfNode* iff, bool flip, b
     if (cd->depends_only_on_test() && _igvn.no_dependent_zero_check(cd)) {
       assert(cd->in(0) == dp, "");
       _igvn.replace_input_of(cd, 0, prevdom);
+      if (range_check_predicate) {
+        // Loads and range check Cast nodes that are control dependent on this range check (that is about to be removed)
```

Comment:
Here we should now be talking about range check predicates, and not just range checks, right?